components/esp_modem/src/esp_modem_dte.cpp

```diff
@@ -125,21 +123,46 @@ bool DTE::set_mode(modem_mode m)
             }
             mode = modem_mode::UNDEF;
             return false;
+        } if (mode == modem_mode::CMUX_MANUAL_MODE) {
+            return true;
         } else {
             mode = m;
             return true;
         }
     }
+    // manual CMUX transitions: Enter CMUX
+    if (m == modem_mode::CMUX_MANUAL_MODE) {
+        if (setup_cmux()) {
+            mode = m;
+            return true;
+        }
+        mode = modem_mode::UNDEF;
+        return false;
+    }
+    // manual CMUX transitions: Exit CMUX
+    if (m == modem_mode::CMUX_MANUAL_EXIT && mode == modem_mode::CMUX_MANUAL_MODE) {
+        if (exit_cmux()) {
+            mode = modem_mode::COMMAND_MODE;
+            return true;
+        }
+        mode = modem_mode::UNDEF;
+        return false;
+    }
+    // manual CMUX transitions: Swap terminals
+    if (m == modem_mode::CMUX_MANUAL_SWAP && mode == modem_mode::CMUX_MANUAL_MODE) {
+        other_term.swap(term);
+        return true;
+    }
     return true;
```

Comment:
The default here is to return `true` if none of the `if` above is selected, shouldn't we have a default case that states that mode is not set? Maybe set to `modem_mode::UNDEF` and return `false`

Reply:
This is the lowest layer of the state machine, so I wanted to let DTE *accept* all possible states. If we refuse some transition here, that upper layers might have trouble switching back (if something goes wrong).

We could maybe make the function return `void` ?

Reply:
My concern with the current version of the code is that it is hard to spot if all modes were covered, and the default makes it that a request to set to a mode that is not covered would not fail. We need at least provide some LOG as diagnostic if that is the case. 

On a quick search of the usage of the function, I see that the return is sometimes discarded and sometimes used, so I think that the error is meaningful, and it should be handled in places where it's discarded ate the moment.

Reply:
Created the default block switching the mode to `modem_mode::UNDEF` and return `false`. Also double-checked the usage, the result of DTE mode switch not to be ignored.